jax/_src/numpy/lax_numpy.py

```diff
@@ -3755,20 +3755,22 @@ def identity(n, dtype=None):
 
 
 @_wraps(np.arange)
-def arange(start, stop=None, step=None, dtype=None):
+def arange(start: core.DimSize, stop: Optional[core.DimSize]=None,
+           step: Optional[core.DimSize]=None, dtype=None):
   lax._check_user_dtype_supported(dtype, "arange")
   require = partial(core.concrete_or_error, _np_asarray)
   msg = "It arose in jax.numpy.arange argument `{}`.".format
+  dtype = dtype or _dtype(start, *(x for x in [stop, step] if x is not None))
```

Comment:
Post-submit: I was trying to incorporate this into the X64 changes, and I'm a bit confused by this line. Below, there is special logic in case `start` is a special dimension handler. But this line (with `_dtype = dtypes.result_type`) fails in this case because `start` cannot be interpreted as a dtype.

What is the correct return type in case `start` is a special dimension handler and `dtype` is not specified?

Reply:
`start` should not be of type `core.DimensionHandler`, it may be though of type `shape_poly._DimPolynomial`.
This should work due to https://github.com/google/jax/blob/639517f97f155fa5c74b449f8c177ddf7b5b9ef1/jax/experimental/jax2tf/shape_poly.py#L453

This case is tested by shape_poly_tests: 

```
pytest --durations=10 -r a --verbosity 1 -s jax/experimental/jax2tf/tests/shape_poly_test.py -k "arange_start_no_dtype"
```